core/domain/opportunity_services.py

```diff
@@ -189,8 +189,65 @@ def add_new_exploration_opportunities(story_id, exp_ids):
     """
     story = story_fetchers.get_story_by_id(story_id)
     topic = topic_fetchers.get_topic_by_id(story.corresponding_topic_id)
-    explorations = exp_fetchers.get_multiple_explorations_by_id(exp_ids)
+    _create_exploration_opportunities(story, topic, exp_ids)
+
+
+def create_exploration_opportunities_for_story(story_id, topic_id):
```

Comment:
Should we use story.corresponding_topic_id and remove the topic_id as a param (this will make it simpler in terms of function name and the corresponding params)

Reply:
I initially tried that, but the upstream call to `publish_story` also takes in both story_id and topic_id, so I've just propagated those arguments here (otherwise, topic_viewer tests fail).